docs/usage/CUSTOMER_SYNC.md

```diff
@@ -69,28 +69,40 @@ Therefore, in order to resolve the actual ids of those references in the sync pr
 
 ````java
-// Build a CustomerQuery for fetching customers from a source CTP project with all the needed references expanded for the sync
-final CustomerQuery customerQueryWithReferenceExpanded = CustomerReferenceResolutionUtils.buildCustomerQuery();
+// Build a CustomerQuery for fetching customers from a source CTP project without any references expanded for the sync:
+final CustomerQuery customerQueryWithReferences = CustomerReferenceResolutionUtils.buildCustomerQuery();
 
 // Query all customers (NOTE this is just for example, please adjust your logic)
 final List<Customer> customers =
     CtpQueryUtils
-        .queryAll(sphereClient, customerQueryWithReferenceExpanded, Function.identity())
+        .queryAll(sphereClient, customerQueryWithReferences, Function.identity())
         .thenApply(fetchedResources -> fetchedResources
             .stream()
             .flatMap(List::stream)
             .collect(Collectors.toList()))
         .toCompletableFuture()
         .join();
+````
+
+In order to transform and map the customer, 
+Initialize [`CustomerReferenceTransformService`](https://github.com/commercetools/commercetools-sync-java/tree/master/src/main/java/com/commercetools/sync/customers/service/CustomerReferenceTransformService.java) with `sphereClient` and cache(You can use your own cache implementation and pass the map).
+For cache implementation, you can refer an example class in the library - which implements the cache using caffeine library with an LRU based cache eviction strategy[`InMemoryReferenceIdToKeyCache`](https://github.com/commercetools/commercetools-sync-java/tree/master/src/main/java/com/commercetools/sync/commons/utils/InMemoryReferenceIdToKeyCache.java).
+Then call the transformCustomerReferences method with the `customers` parameter as shown below:
 
-// Mapping from Customer to CustomerDraft with considering reference resolution.
-final List<CustomerDraft> customerDrafts = CustomerReferenceResolutionUtils.mapToCustomerDrafts(customers);
+````java
+// Fetch(Id to key values for references) into the cache and map from Customer to CustomerDraft using cache with considering reference resolution.
+final List<CustomerDraft> customerDrafts = CustomerReferenceTransformService.transformCustomerReferences(customers);
 ````
 
+The cache here is used for a better performance. 
+Instead of expanding the references in the query for customer resource. `CategoryReferenceTransformService` will execute a query to fetch key values for the Ids(nonCached) and store in cache. These cached id to key values then can be used by another resource for resolving its references instead of fetching from commercetools API. It turns out, having the in-memory LRU cache will improve the overall performance of the sync library and commercetools API.
```

Comment:
```suggestion
Instead of expanding the references in the query for customer resource. `CategoryReferenceTransformService` will execute a query to fetch key-id pairs and store in cache. These cached id to key values then can be used by another resource for resolving its references instead of fetching from commercetools API. It turns out, having the in-memory LRU cache will improve the overall performance of the sync library and commercetools API.
```